### ISSUE TITLE
[Bug] wallet startup failure 

### ISSUE THREAD

**Issue by 0x5BC2E7** [Chia-Network/chia-blockchain#14888]
### What happened?

```
2023-03-23T11:52:48.117 wallet wallet                     : INFO     Starting service wallet ...
2023-03-23T11:52:48.118 wallet wallet                     : INFO     chia-blockchain version: 1.7.1
2023-03-23T11:52:48.118 wallet wallet_server              : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T11:55:17.604 daemon chia.daemon.server         : INFO     ConnectionClosed. Closing websocket with ['Unknown']
2023-03-23T11:55:18.608 wallet wallet                     : INFO     Starting service wallet ...
2023-03-23T11:55:18.609 wallet wallet                     : INFO     chia-blockchain version: 1.7.1
2023-03-23T11:55:18.609 wallet wallet_server              : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T11:58:23.987 daemon chia.daemon.server         : INFO     ConnectionClosed. Closing websocket with ['Unknown']
2023-03-23T11:58:24.858 harvester chia.harvester.harvester: INFO     `harvester.plot_loading_frequency_seconds` is deprecated. Consider replacing it with the new section `harvester.plots_refresh_parameter`. See `initial-config.yaml`.
2023-03-23T11:58:24.859 harvester chia.harvester.harvester: INFO     Using plots_refresh_parameter: {'batch_size': 300, 'batch_sleep_milliseconds': 1, 'interval_seconds': 120, 'retry_invalid_seconds': 1200}
2023-03-23T11:58:24.859 harvester harvester               : INFO     Starting service harvester ...
2023-03-23T11:58:24.859 harvester harvester               : INFO     chia-blockchain version: 1.7.1
2023-03-23T11:58:24.860 harvester harvester_server        : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T11:58:24.865 harvester harvester               : INFO     Started harvester service on network_id: mainnet at port 8448
2023-03-23T11:58:24.868 harvester harvester               : INFO     Reconnecting to peer PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8447)
2023-03-23T11:58:24.871 harvester harvester_server        : INFO     Cannot connect to host 127.0.0.1:8447 ssl:<ssl.SSLContext object at 0x7f5974c47c40> [Connect call failed ('127.0.0.1', 8447)]
2023-03-23T11:58:24.883 daemon chia.daemon.server         : INFO     Register service {'service': 'chia_harvester'}
2023-03-23T11:58:24.884 daemon chia.daemon.server         : INFO     registered for service chia_harvester
2023-03-23T11:58:24.884 daemon chia.daemon.server         : INFO     {'success': True}
2023-03-23T11:58:25.011 farmer farmer                     : INFO     Starting service farmer ...
2023-03-23T11:58:25.011 farmer farmer                     : INFO     chia-blockchain version: 1.7.1
2023-03-23T11:58:25.011 farmer farmer_server              : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T11:58:25.059 farmer farmer_server              : INFO     Started listening on port: 8447
2023-03-23T11:58:25.060 farmer farmer                     : INFO     Started farmer service on network_id: mainnet at port 8447
2023-03-23T11:58:25.062 farmer farmer                     : INFO     Reconnecting to peer PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8444)
2023-03-23T11:58:25.065 farmer farmer_server              : INFO     Cannot connect to host 127.0.0.1:8444 ssl:<ssl.SSLContext object at 0x7f70925f3840> [Connect call failed ('127.0.0.1', 8444)]
2023-03-23T11:58:25.077 daemon chia.daemon.server         : INFO     Register service {'service': 'chia_farmer'}
2023-03-23T11:58:25.078 daemon chia.daemon.server         : INFO     registered for service chia_farmer
2023-03-23T11:58:25.078 daemon chia.daemon.server         : INFO     {'success': True}
2023-03-23T11:58:25.157 full_node full_node               : INFO     Starting service full_node ...
2023-03-23T11:58:25.158 full_node full_node               : INFO     chia-blockchain version: 1.7.1
2023-03-23T11:58:25.158 full_node full_node_server        : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T11:58:25.161 full_node chia.full_node.full_node: INFO     Default port field not found in config.
2023-03-23T11:58:25.182 full_node chia.full_node.full_node: ERROR    Exception: [Errno -2] Name or service not known
2023-03-23T11:58:25.182 full_node chia.full_node.full_node: ERROR    Exception in peer discovery: [Errno -2] Name or service not known
2023-03-23T11:58:25.183 full_node chia.full_node.full_node: ERROR    Exception Stack: Traceback (most recent call last):
File "chia/full_node/full_node.py", line 496, in set_server
File "chia/server/node_discovery.py", line 531, in __init__
File "chia/server/node_discovery.py", line 68, in __init__
File "chia/util/network.py", line 151, in get_host_addr
File "socket.py", line 918, in getaddrinfo
socket.gaierror: [Errno -2] Name or service not known
```

### Version

1.7.1

### What platform are you using?

Linux

### What ui mode are you using?

CLI

### Relevant log output

```shell
File "chia/server/start_wallet.py", line 118, in <module>
File "chia/server/start_wallet.py", line 114, in main
File "chia/server/start_service.py", line 283, in async_run
File "asyncio/runners.py", line 44, in run
File "asyncio/base_events.py", line 616, in run_until_complete
File "chia/server/start_wallet.py", line 103, in async_main
File "chia/server/start_wallet.py", line 68, in create_wallet_service
File "chia/server/start_service.py", line 117, in __init__
File "chia/wallet/wallet_node.py", line 613, in set_server
File "chia/wallet/wallet_node.py", line 622, in initialize_wallet_peers
File "chia/server/node_discovery.py", line 711, in __init__
File "chia/server/node_discovery.py", line 68, in __init__
File "chia/util/network.py", line 151, in get_host_addr
File "socket.py", line 918, in getaddrinfo
socket.gaierror: [Errno -2] Name or service not known
[2660629] Failed to execute script 'start_wallet' due to unhandled exception!
```

**Comment by 0x5BC2E7** [comment]
`chia start farmer`

**Comment by 0x5BC2E7** [comment]
Rolled back to 1.7.0 - everything works fine as it was before

**Comment by ZeroCool1974-BR** [comment]
> Rolled back to 1.7.0 - everything works fine as it was before

had the same problem with Win10_x64, constant sync/not sync issues and returned to safety with 1.7.0

**Comment by MumfMeisterT** [comment]
Could you please attach the full node section of your config.yaml file for us to review? 
After which you can delete and regenerate your config.yaml file and this should resolve.

**Comment by MumfMeisterT** [comment]
> > Rolled back to 1.7.0 - everything works fine as it was before
> 
> had the same problem with Win10_x64, constant sync/not sync issues and returned to safety with 1.7.0

Could you please provide more info around this? I believe this is a separate issue, but we would like to hear more about this. Do you have an error log we could get from you? thx.

**Comment by emlowe** [comment]
Thank you for the report - yes it looks like we introduced this in 1.7.1.

The code here: 
```  if introducer_info is not None:
            # get_host_addr is blocking but this only gets called on startup or in the wallet after disconnecting from
            # all trusted peers.
            self.introducer_info: Optional[PeerInfo] = PeerInfo(
                str(get_host_addr(introducer_info["host"], prefer_ipv6=False)),
                introducer_info["port"],
            )```
            
  doesn't handle cases where `get_host_addr` raises/throws which will happen when the name cannot be resolved

**Comment by emlowe** [comment]
@0x5BC2E7 A workaround for you is to remove the introducer line from config.yaml and I think it will work for you. Remove this section 
```
  introducer_peer:
    host: introducer.chia.net # Chia AWS introducer IPv4/IPv6
    port: 8444
```

and I believe it will bypass this code

**Comment by 0x5BC2E7** [comment]
> @0x5BC2E7 A workaround for you is to remove the introducer line from config.yaml and I think it will work for you. Remove this section
> 
> ```
>   introducer_peer:
>     host: introducer.chia.net # Chia AWS introducer IPv4/IPv6
>     port: 8444
> ```
> 
> and I believe it will bypass this code

Done it with wallet/full_node sections - stuck on init, no sync. Rolled back to 1.7.0 :(

`2023-03-23T22:02:26.547 daemon chia.daemon.server         : INFO     ConnectionClosed. Closing websocket with ['Unknown']
2023-03-23T22:02:27.280 harvester chia.harvester.harvester: INFO     `harvester.plot_loading_frequency_seconds` is deprecated. Consider replacing it with the new section `harvester.plots_refresh_parameter`. See `initial-config.yaml`.
2023-03-23T22:02:27.280 harvester chia.harvester.harvester: INFO     Using plots_refresh_parameter: {'batch_size': 300, 'batch_sleep_milliseconds': 1, 'interval_seconds': 120, 'retry_invalid_seconds': 1200}
2023-03-23T22:02:27.281 harvester harvester               : INFO     Starting service harvester ...
2023-03-23T22:02:27.281 harvester harvester               : INFO     chia-blockchain version: 1.7.1
2023-03-23T22:02:27.281 harvester harvester_server        : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T22:02:27.285 harvester harvester               : INFO     Started harvester service on network_id: mainnet at port 8448
2023-03-23T22:02:27.287 harvester harvester               : INFO     Reconnecting to peer PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8447)
2023-03-23T22:02:27.289 harvester harvester_server        : INFO     Cannot connect to host 127.0.0.1:8447 ssl:<ssl.SSLContext object at 0x7f9b513e4c40> [Connect call failed ('127.0.0.1', 8447)]
2023-03-23T22:02:27.300 daemon chia.daemon.server         : INFO     Register service {'service': 'chia_harvester'}
2023-03-23T22:02:27.300 daemon chia.daemon.server         : INFO     registered for service chia_harvester
2023-03-23T22:02:27.301 daemon chia.daemon.server         : INFO     {'success': True}
2023-03-23T22:02:27.620 farmer farmer                     : INFO     Starting service farmer ...
2023-03-23T22:02:27.621 farmer farmer                     : INFO     chia-blockchain version: 1.7.1
2023-03-23T22:02:27.621 farmer farmer_server              : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T22:02:27.666 farmer farmer_server              : INFO     Started listening on port: 8447
2023-03-23T22:02:27.667 farmer farmer                     : INFO     Started farmer service on network_id: mainnet at port 8447
2023-03-23T22:02:27.668 farmer farmer                     : INFO     Reconnecting to peer PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8444)
2023-03-23T22:02:27.671 farmer farmer_server              : INFO     Cannot connect to host 127.0.0.1:8444 ssl:<ssl.SSLContext object at 0x7f3d10a82840> [Connect call failed ('127.0.0.1', 8444)]
2023-03-23T22:02:27.763 full_node full_node               : INFO     Starting service full_node ...
2023-03-23T22:02:27.764 full_node full_node               : INFO     chia-blockchain version: 1.7.1
2023-03-23T22:02:27.764 full_node full_node_server        : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T22:02:27.767 full_node chia.full_node.full_node: INFO     Default port field not found in config.
2023-03-23T22:02:27.778 full_node chia.full_node.full_node: ERROR    Exception: [Errno -2] Name or service not known
2023-03-23T22:02:27.779 full_node chia.full_node.full_node: ERROR    Exception in peer discovery: [Errno -2] Name or service not known
2023-03-23T22:02:27.779 full_node chia.full_node.full_node: ERROR    Exception Stack: Traceback (most recent call last):
File "chia/full_node/full_node.py", line 496, in set_server
File "chia/server/node_discovery.py", line 531, in __init__
File "chia/server/node_discovery.py", line 68, in __init__
File "chia/util/network.py", line 151, in get_host_addr
File "socket.py", line 918, in getaddrinfo
socket.gaierror: [Errno -2] Name or service not known

2023-03-23T22:02:27.783 full_node chia.full_node.full_node: INFO     using blockchain database /home/chia/.chia/mainnet/db/blockchain_v2_mainnet.sqlite, which is version 2
2023-03-23T22:02:27.784 full_node chia.full_node.full_node: INFO     opening blockchain DB: synchronous=FULL
2023-03-23T22:02:27.790 daemon chia.daemon.server         : INFO     Register service {'service': 'chia_farmer'}
2023-03-23T22:02:27.790 daemon chia.daemon.server         : INFO     registered for service chia_farmer
2023-03-23T22:02:27.790 full_node chia.full_node.block_store: INFO     DB: Creating block store tables and indexes.
2023-03-23T22:02:27.791 daemon chia.daemon.server         : INFO     {'success': True}
2023-03-23T22:02:27.791 full_node chia.full_node.block_store: INFO     DB: Creating index height
2023-03-23T22:02:27.792 full_node chia.full_node.block_store: INFO     DB: Creating index is_fully_compactified
2023-03-23T22:02:27.792 full_node chia.full_node.block_store: INFO     DB: Creating index main_chain
2023-03-23T22:02:27.793 full_node chia.full_node.hint_store: INFO     DB: Creating hint store tables and indexes.
2023-03-23T22:02:27.794 full_node chia.full_node.hint_store: INFO     DB: Creating index hint_index
2023-03-23T22:02:27.795 full_node chia.full_node.coin_store: INFO     DB: Creating coin store tables and indexes.
2023-03-23T22:02:27.795 full_node chia.full_node.coin_store: INFO     DB: Creating index coin_confirmed_index
2023-03-23T22:02:27.795 full_node chia.full_node.coin_store: INFO     DB: Creating index coin_spent_index
2023-03-23T22:02:27.796 full_node chia.full_node.coin_store: INFO     DB: Creating index coin_puzzle_hash
2023-03-23T22:02:27.796 full_node chia.full_node.coin_store: INFO     DB: Creating index coin_parent_index
2023-03-23T22:02:27.796 full_node chia.full_node.full_node: INFO     Initializing blockchain from disk
2023-03-23T22:02:27.796 full_node chia.full_node.full_node: INFO     Selected multiprocessing start method: default
2023-03-23T22:02:27.798 full_node chia.consensus.blockchain: INFO     Started 8 processes for block validation
2023-03-23T22:02:27.859 wallet wallet                     : INFO     Starting service wallet ...
2023-03-23T22:02:27.859 wallet wallet                     : INFO     chia-blockchain version: 1.7.1
2023-03-23T22:02:27.859 wallet wallet_server              : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2023-03-23T22:02:28.163 farmer chia.farmer.farmer         : INFO     Added pool: {'launcher_id': '0xXXX',
'owner_public_key': '0xXXX',
'p2_singleton_puzzle_hash': '0xXXX',
'payout_instructions': '0XXX',
'pool_url': 'https://eu1.pool.space',
'target_puzzle_hash': '0x2f2c9ba1b2315d413a92b5f034fa03282ccba1767fd9ae7b14d942b969ed5d57'}
2023-03-23T22:02:28.407 full_node chia.full_node.full_node: INFO     Blockchain initialized to peak 9cee0604f088c45d8e6853b9aaa94e7476da8a5e4f181259954b3422cc48ad60 height 3420644, time taken: 0s
2023-03-23T22:02:28.427 full_node chia.full_node.full_node: INFO     🌱 Updated peak to height 3420644, weight 6991213856, hh 9cee0604f088c45d8e6853b9aaa94e7476da8a5e4f181259954b3422cc48ad60, forked at 3420643, rh: 2a000a140b0e44370e39e682189d773e6a9b4a9df73239969a92beda21131def, total iters: 14176456884471, overflow: False, deficit: 0, difficulty: 1960, sub slot iters: 143654912, Generator size: No tx, Generator ref list size: No tx
2023-03-23T22:02:28.441 full_node full_node_server        : INFO     Started listening on port: 8444
2023-03-23T22:02:28.441 full_node full_node               : INFO     Started full_node service on network_id: mainnet at port 8444
2023-03-23T22:02:28.457 daemon chia.daemon.server         : INFO     Register service {'service': 'chia_full_node'}
2023-03-23T22:02:28.457 daemon chia.daemon.server         : INFO     registered for service chia_full_node
2023-03-23T22:02:28.457 daemon chia.daemon.server         : INFO     {'success': True}
2023-03-23T22:02:28.507 farmer chia.farmer.farmer         : INFO     GET /pool_info response: {'name': 'Space Pool', 'logo_url': 'https://pool.space/logo-250.png', 'minimum_difficulty': 1, 'relative_lock_height': 64, 'protocol_version': 1, 'fee': 0.01, 'description': 'Welcome to Space Pool. By creating this Plot NFT, you agree to the following terms https://pool.space/terms', 'target_puzzle_hash': '0xXXX', 'authentication_token_timeout': 5}
2023-03-23T22:02:28.830 farmer chia.farmer.farmer         : INFO     GET /farmer response: {'authentication_public_key': '0xXXX', 'payout_instructions': '0XXX', 'current_difficulty': 15, 'current_points': 1827}
2023-03-23T22:02:30.293 harvester harvester               : INFO     Reconnecting to peer PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8447)
2023-03-23T22:02:30.308 harvester harvester_server        : INFO     Connected with farmer PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8447)
2023-03-23T22:02:30.324 harvester chia.plotting.cache     : INFO     Loaded 719068 bytes of cached data
2023-03-23T22:02:30.450 harvester chia.plotting.cache     : INFO     Parsed 730 cache entries in 0.13s
2023-03-23T22:02:30.509 harvester chia.harvester.harvester: INFO     _plot_refresh_callback: event started, loaded 0, removed 0, processed 0, remaining 730, duration: 0.00 seconds, total plots: 0
2023-03-23T22:02:30.555 farmer chia.plot_sync.receiver    : INFO     reset: node_id dc7421296f4a8191bca8e56cbdeb7e69df4f283f2bc6a40c24b98bf7d4ac705a, current_sync: [state 0, sync_id 0, next_message_id 0, plots_processed 0, plots_total 0, delta [valid +0/-0, invalid +0/-0, keys missing: +0/-0, duplicates: +0/-0], time_done None]
2023-03-23T22:02:30.694 harvester chia.harvester.harvester: INFO     _plot_refresh_callback: event done, loaded 730, removed 0, processed 730, remaining 0, duration: 0.16 seconds, total plots: 730
2023-03-23T22:02:30.711 farmer farmer                     : INFO     Reconnecting to peer PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8444)
2023-03-23T22:02:30.723 farmer farmer_server              : INFO     Connected with full_node PeerInfo(_ip=IPv4Address('127.0.0.1'), _port=8444)`

**Comment by emlowe** [comment]
Second workaround, add into `/etc/hosts` an entry for `100.21.23.17 introducer.chia.net`. But yes, rollback to 1.7.0 is also ok until we fix this issue

**Comment by 0x5BC2E7** [comment]
Good news - it was fixed by changing `introducer-eu.chia.net` >> `introducer.chia.net`. Hope it will helps somebody

**Comment by emlowe** [comment]
Glad you found a workaround - re-opening this because it is still a bug that if the introducer doesn't resolve, the services don't start and the code should be resilient to that

**Comment by emlowe** [comment]
Fixed in 1.8.0
